### PR TITLE
udiskie: add sni support

### PR DESCRIPTION
**Pull request by pltanton**
Adds support of `--appindicator` flag, that starts `udiskie` in sni compatible mode.

### CODE REVIEW

File: modules/services/udiskie.nix [hunk 1]
```diff
@@ -73,7 +79,7 @@ in
         };
 
         Service = {
-          ExecStart = "${pkgs.udiskie}/bin/udiskie -2 ${commandArgs}";
+          ExecStart = "${pkgs.udiskie}/bin/udiskie -2 ${commandArgs} ${if cfg.sni then "--appindicator" else ""}";
```
Comment (teto) [inline modules/services/udiskie.nix]: why put it there instead of the commandArgs. What is Sni ?

Comment (teto) [reply modules/services/udiskie.nix]: I guess "status-notifier-item" from your toher pr :)

Comment (pltanton) [reply modules/services/udiskie.nix]: commandArgs is generated with uniform mapping which adds unary minus to options. I'm not sure where it should be placed, but it seems like that you are right and it should be in commandArgs just by concatenating.

FYI. Sni is a standard for "modern tray" that comes after classical systray it used in KDE, gnome and taffybar f.e.

**Comment by rycee** [comment]
Thanks! I've rebased this to master in 29ad012763f4e07c5c50635915c1d4ae751123cc. Note, I changed it slightly to use the `optional` function instead.

**Pull request closed without merging**